boa/src/syntax/ast/node/object.rs

```diff
@@ -0,0 +1,88 @@
+//! Object node.
+
+use super::Node;
+use gc::{Finalize, Trace};
+use std::fmt;
+
+use crate::syntax::ast::node::PropertyDefinition;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// Objects in JavaScript may be defined as an unordered collection of related data, of
+/// primitive or reference types, in the form of “key: value” pairs.
+///
+/// Objects can be initialized using `new Object()`, `Object.create()`, or using the literal
+/// notation.
+///
+/// An object initializer is an expression that describes the initialization of an
+/// [`Object`][object]. Objects consist of properties, which are used to describe an object.
+/// Values of object properties can either contain [`primitive`][primitive] data types or other
+/// objects.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-ObjectLiteral
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer
+/// [object]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
+/// [primitive]: https://developer.mozilla.org/en-US/docs/Glossary/primitive
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct Object {
+    properties: Box<[PropertyDefinition]>,
+}
+
+impl Object {
+    /// Creates an `Object` AST node.
+    pub fn new<D>(props: D) -> Self
```

Comment:
Maybe we should implement the `From` trait instead of the `new()` function. How do we do it for other "list" type nodes?

Reply:
Is the reason for this that From is more useful / can be used in more situations?

Reply:
> Is the reason for this that From is more useful / can be used in more situations?

It's just an standard trait, that can help on function signatures making them more generic, if needed.

Reply:
Ok cool, changed :)